src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -79,10 +77,16 @@ public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeP
 
                     if (logEntry.State is IReadOnlyCollection<KeyValuePair<string, object>> stateDictionary)
                     {
+                        writer.WriteStartObject(nameof(logEntry.State));
                         foreach (KeyValuePair<string, object> item in stateDictionary)
                         {
-                            writer.WriteString(item.Key, Convert.ToString(item.Value, CultureInfo.InvariantCulture));
+                            writer.WriteString(item.Key, ToInvariantString(item.Value));
                         }
+                        writer.WriteEndObject();
+                    }
+                    else if (logEntry.State != null)
+                    {
+                        writer.WriteString(nameof(logEntry.State), logEntry.State.ToString());
```

Comment:
> Also note the current implementation is missing the rendered message for some of the scopes, not sure if that should be tracked as separate issue?

(taken from https://github.com/dotnet/runtime/issues/39664#issuecomment-662315321)
@alefranz I think this is what you were asking for right?

Reply:
Not really, I was referring to the scope. Sorry I should have raised the issue before. Here it is: https://github.com/dotnet/runtime/issues/39867

Regarding this change, in which scenarios would the state not be a k/v collection?

Reply:
> Regarding this change, in which scenarios would the state not be a k/v collection?

Refer to test: Log_IncludeScopes_ContainsDuplicateNamedPropertiesInScope_AcceptableJson 

```c#
            using (logger.BeginScope("scope1 {name1}", 123))
            using (logger.BeginScope("scope2 {name1} {name2}", 456, 789))
                logger.Log(LogLevel.Information, 0, state: "exception message", exception: null, formatter: (a, b) => a);
```